src/IO/ParallelReadBuffer.cpp

```diff
@@ -48,8 +48,15 @@ ParallelReadBuffer::ParallelReadBuffer(
     , max_working_readers(max_working_readers_)
     , schedule(std::move(schedule_))
     , reader_factory(std::move(reader_factory_))
+{}
```

Comment:
That is true, we have a new dangling state.
The alternative is to add implicit initialisation on first seek/next (still needs to be added to both places) which I don't like because the buffering itself will only start on the first call to seek/next.
We can also catch the exceptions in the constructor but if we want to throw it, we need to throw it on calls of next/seek.
To me, this gives the most control and if someone wants to initialise the buffer at a later point he can do that. If you forget, you get a straightforward error IMO.